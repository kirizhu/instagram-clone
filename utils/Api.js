import axios from "axios";

axios.defaults.baseURL = "https://picsum.photos";

export const fetchData = async ({ url }) => {
  let result = null;
  let errorMessage = "";
  let loading = true;
  try {
    const { data } = await axios.get(url);
    result = data;
  } catch (error) {
    errorMessage = error;
  } finally {
    loading = false;
    return { result, errorMessage, loading };
  }
};
