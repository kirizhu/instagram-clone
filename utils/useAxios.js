import { useState, useEffect } from "react";
import axios from "axios";
import fetchData from "./Api";

axios.defaults.baseURL = "https://picsum.photos";

const useAxios = ({ url }) => {
  const [response, setResponse] = useState(null);
  const [error, setError] = useState("");
  const [loading, setloading] = useState(true);

  const fetchData = async () => {
    try {
      let res = await axios.get(url);
      setResponse(res.data);
    } catch (err) {
      setError(err);
    } finally {
      setloading(false);
    }
  };

  useEffect(() => {
    fetchData();
  }, [url]);

  return { response, error, loading };
};

export default useAxios;
