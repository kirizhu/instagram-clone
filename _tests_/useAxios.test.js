import { renderHook } from "@testing-library/react-hooks";
import useAxios from "../utils/useAxios";
import { fetchData } from "../utils/Api";

describe("fetchData", () => {
  test("should return an object", async () => {
    // const { result } = await renderHook(() => {
    //   useAxios({ url: "/id/0/info" });
    // });
    const res = await fetchData({ url: "/id/0/info" });
    console.log(res);
    const expected = {
      id: "0",
      author: "Alejandro Escamilla",
      width: 5616,
      height: 3744,
      url: "https://unsplash.com/photos/yC-Yzbqy7PY",
      download_url: "https://picsum.photos/id/0/5616/3744",
    };
    expect(res.result).toMatchSnapshot();
    expect(res.result).toEqual(expected);
  });
  test("should return an array of 30 in length", async () => {
    const res = await fetchData({ url: `/v2/list` });

    const expected = 30;
    expect(res.result).toHaveLength(expected);
  });
  test("Objects in array should have certain properties", async () => {
    const res = await fetchData({ url: `/v2/list` });
    res.result.forEach((item) => {
      expect(item).toHaveProperty("id");
      expect(item).toHaveProperty("author");
      expect(item).toHaveProperty("width");
      expect(item).toHaveProperty("height");
      expect(item).toHaveProperty("url");
      expect(item).toHaveProperty("download_url");
    });
  });
});
