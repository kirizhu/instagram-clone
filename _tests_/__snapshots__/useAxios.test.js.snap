// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fetchData should return an object 1`] = `
Object {
  "author": "Alejandro Escamilla",
  "download_url": "https://picsum.photos/id/0/5616/3744",
  "height": 3744,
  "id": "0",
  "url": "https://unsplash.com/photos/yC-Yzbqy7PY",
  "width": 5616,
}
`;
